nessclient/event.py

```diff
@@ -32,11 +33,13 @@ def __repr__(self) -> str:
         return "<{} {}>".format(self.__class__.__name__, self.__dict__)
 
     @classmethod
-    def decode(cls, packet: Packet) -> "BaseEvent":
+    def decode(cls, packet: Packet) -> Union["BaseEvent", CommandType]:
```

Comment:
Would it be possible to avoid this type-union in this instance? I'd consider this an antipattern - `BaseEvent::decode` should only ever return (child) instances of `BaseEvent`, not anything else.

Ideally we should avoid calling `BaseEvent::decode` entirely if we know `CommandType == CommandType.TEST_ASCII_STRING` in the parent caller.

```suggestion
    def decode(cls, packet: Packet) -> "BaseEvent":
```